### DISCUSSION TITLE
What all time series models does rivier support?

### DISCUSSION THREAD

**Discussion by sushanthsandu006** [online-ml/river#605] [Q&A] [answered]
Is there only SNARIMAX model that can be used?

**Comment by raphaelsty** [comment, marked as answer]
Hi, you can also use neural networks, tree based models, linear models to perform time series forecasting. 

[Here](https://machinelearningmastery.com/random-forest-for-time-series-forecasting/) is a tutorial to construct a dataset for time series forecasting.

**Reply by sushanthsandu006** [reply]
@raphaelsty But river supports only snarimax.

**Reply by raphaelsty** [reply]
Here are some links to the documentation, those models can be used for time series forecasting:

- [neural networks](https://riverml.xyz/latest/api/neural-net/MLPRegressor/)
- [trees](https://riverml.xyz/latest/api/tree/HoeffdingAdaptiveTreeRegressor/)
- [linear](https://riverml.xyz/latest/api/linear-model/LinearRegression/)
